core-bundle/src/Slug/Slug.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Slug;
+
+use Ausi\SlugGenerator\SlugGenerator;
+use Contao\CoreBundle\Framework\ContaoFrameworkInterface;
+use Contao\PageModel;
+use Contao\StringUtil;
+
+class Slug
+{
+    /**
+     * @var SlugGenerator
+     */
+    private $slugGenerator;
+
+    /**
+     * @var ContaoFrameworkInterface
+     */
+    private $framework;
+
+    public function __construct(SlugGenerator $slugGenerator, ContaoFrameworkInterface $framework)
+    {
+        $this->slugGenerator = $slugGenerator;
+        $this->framework = $framework;
+    }
+
+    /**
+     * @param int|iterable $options        A page ID, object or options array {@see SlugGenerator::generate()}
+     * @param callable     $duplicateCheck A callback to check if the slug is already in use
+     */
+    public function generate(string $text, $options = [], callable $duplicateCheck = null, string $integerPrefix = 'id-'): string
+    {
+        if (!is_iterable($options)) {
+            /** @var PageModel $pageAdapter */
+            $pageAdapter = $this->framework->getAdapter(PageModel::class);
+
+            if (null !== ($page = $pageAdapter->findWithDetails((int) $options))) {
```

Comment:
This feels a bit weird to me. Can we not make this `@param iterable` only and pass `['pageId' => 4]` instead of an integer?

Reply:
What would then happen if you pass something like `['pageId' => 4, 'validChars' => '0-9a-z']`?

IMO the current way is easier to distinguish: pass an integer if you want to get the options from a page ID, otherwise pass the options themselfs.

Reply:
I would interpret this as "use the slug options from page ID 4 but overwrite the `validChars` setting with a custom value". This seems even more flexible to me actually, don't you think?